Fix -Wimplicit-function-declaration error on a test with gcc 14.

Error:

```
run_test.c: In function 'wait_for':
run_test.c:93:21: error: implicit declaration of function 'strcasestr'; did you
mean 'strcasecmp'? [-Wimplicit-function-declaration]
   93 |         while (0 == strcasestr(reply, event)) {
      |                     ^~~~~~~~~~
      |                     strcasecmp
```

--- speech-dispatcher-0.11.5-origin/src/tests/run_test.c
+++ speech-dispatcher-0.11.5/src/tests/run_test.c
@@ -25,6 +25,7 @@
 #endif
 
 #include <stdio.h>
+#define _GNU_SOURCE
 #include <string.h>
 #include <sys/types.h>
 #include <sys/socket.h>
